README.rst

```diff
@@ -26,6 +26,22 @@ This Open edX plugin, ``openedx-completion-aggregator`` watches course activity
 * What % of each section in Course X have my students completed?
 * What is the average completion % among all enrolled students in a course?
 
+Notes:
+
+* This service only provides data, via a REST API. There is no user interface.
+* On production instances, the answers to these "aggregate" questions may be slightly out of date, because they are computed asynchronously (see below). How often they are updated is configurable.
+
+Synchronous vs. Asynchronous calculations
+-----------------------------------------
+
+openedx-completion-aggregator operates in one of two modes: synchronous or asynchronous.
+
+With synchronous aggregation, each time a student completes a block, the aggregator code will re-calculate the aggregate completion values immediately. You will always have the freshest results from this API, but at a huge performance cost. Synchronous aggregation is only for development purposes and is not suitable for production. Synchronous aggregation can cause deadlocks when users complete XBlocks, leading to a partial outage of the LMS.
+
+With asynchronous aggregation, the aggregator code will re-calculate the aggregate completion values asynchronously, at periodic intervals (e.g. every hour). How often the update can and should be run depends on many factors - you will have to experiment and find what works best and what is possible for your specific Open edX installation.
```

Comment:
@bradenmacdonald, we could also add a warning that running this too often can clog the tasks queue, which might require manual intervention.